src/arrayMethodJoin.js

```diff
@@ -4,9 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (const element of this) {
+      switch (true) {
+        case typeof element === 'undefined'
+        && this.indexOf(element) === this.length - 1:
+          result += '';
+          break;
+
+        case this.indexOf(element) === this.length - 1:
+          result += element;
+          break;
+
+        case element === null || typeof element === 'undefined' :
+          result += '' + separator;
+          break;
+
+        default:
+          result += element + '' + separator;
```

Comment:
you don't need this blank string here, like why do you even used it in the first place?